src/main/resources/public/jsx/geo-dash-widget-editor.js

```diff
@@ -605,48 +622,50 @@ class BasicLayout extends React.PureComponent{
                 widget.dualEnd = this.state.endDate2;
             }
         }
-        const holdRef = this;
-
-        $.ajax({
-            url: theURL + "/createwidget/widget",
-            type: "get",
-            dataType: "jsonp",
-            widget: JSON.stringify(widget),
-            data: {
-                pID: pid,
-                dashID: this.state.dashboardID,
-                widgetJSON: JSON.stringify(widget)
-            },
-            success: function () {
-                const myWidget = JSON.parse(this.widget);
-                holdRef.setState({
-                    widgets: holdRef.state.widgets.concat(myWidget),
-                    selectedWidgetType: "-1",
-                    selectedDataTypeDual: "-1",
-                    isEditing: false,
-                    selectedDataType: "-1",
-                    WidgetTitle: "",
-                    imageCollection: "",
-                    graphBand: "",
-                    graphReducer: "Min",
-                    imageParams: "",
-                    WidgetBaseMap: "osm",
-                    dualLayer: false,
-                    startDate:"",
-                    endDate:"",
-                    startDate2:"",
-                    endDate2:"",
-                    widgetBands:"",
-                    widgetMin:"",
-                    widgetMax:"",
-                    widgetCloudScore:"",
-                    FormReady: false
-                });
-            },
-            error: function (xhr) {
-            }
-        });
 
+        fetch(this.props.theURI + "/createwidget/widget",
+              {
+                  method: "post",
+                  headers: {
+                      "Accept": "application/json",
+                      "Content-Type": "application/json"
+                  },
+                  body: JSON.stringify({
+                      pID: this.state.pid,
+                      dashID: this.state.dashboardID,
+                      widgetJSON: JSON.stringify(widget)
+                  })
+              })
+            .then(response => {
+                if (response.ok){
+                    this.setState({
+                        widgets: this.state.widgets.concat(widget),
```

Comment:
dont mutate the state.  use the spread operator to create a new value [...this.state.widgets, widget]